pxd.c

```diff
@@ -1694,9 +1708,38 @@ ssize_t pxd_export(struct fuse_conn *fc, uint64_t dev_id)
 static void pxd_finish_remove(struct work_struct *work)
 {
 	struct pxd_device *pxd_dev = container_of(work, struct pxd_device, remove_work);
```

Comment:
entire logic within this function seems wrong. This function cannot return early, it has to complete device cleanup